osutil/epoll/epoll.go

```diff
@@ -149,11 +149,27 @@ func (e *Epoll) WaitTimeout(duration time.Duration) ([]Event, error) {
 	if duration < 0 {
 		msec = -1
 	}
-	sysEvents := make([]unix.EpollEvent, e.registeredFdCount)
-	n, err := unix.EpollWait(e.fd, sysEvents, msec)
-	runtime.KeepAlive(e)
-	if err != nil {
-		return nil, err
+	n := 0
+	var err error
+	var sysEvents []unix.EpollEvent
+	for {
+		sysEvents = make([]unix.EpollEvent, e.registeredFdCount)
+		startTs := time.Now()
+		n, err = unix.EpollWait(e.fd, sysEvents, msec)
+		runtime.KeepAlive(e)
+		if err == nil {
```

Comment:
(super-nitpick) I think a `switch {` looks nicer for cases (no pun intended) like this:
```go
// epoll_wait can return EINTR, then we need to adjust the timeout and restart the syscall (if there is one)
switch {
    case err == nil: 
        break
    case err != unix.EINTR: 
        return nil, err
    case msec == -1:
        continue
}
```
and maybe a small comment at the top?

Reply:
I considered using a switch statement here, but breaking out of a for loop from within a switch would require an additional label outside the `for` statement, for which the semantics are still unintuitive to me (I'd use a goto but.... I think not).

Adding a comment is a good idea, I'll do so. For now I'll leave the if else if statements. I think they could be more legible split into separate `if { }; if { }; if { }` rather than `if { } else if ...`, but this was a few fewer lines of code.

Feel free to `sudo !!` if you'd like :)